src/components/calendar-view/CalendarView.tsx

```diff
@@ -397,14 +388,21 @@ function Component({
     };
   }, [calendarRef, view, hasFocus, monthInView]);
 
+  useEffect(() => {
+    if (isValid(viewDate)) {
+      setMonthInView(viewDate.getMonth() as MonthIndices);
+      setYearInView(viewDate.getFullYear());
+    }
+  }, [viewDate]);
```

Comment:
As I mentioned in my previous comment. If the user gives us a viewDate in a non-optimized way this will be triggered for every render resulting in weird behaviour. That's why it was converted to string in the parent component.